polartools/diffraction.py

```diff
@@ -350,3 +350,274 @@ def fit_series(
+                dataz[index] = table[detector] / table[monitor]
+            else:
+                dataz[index] = table[detector]
+
+            if log:
+                for var in range(0, len(dataz[index])):
+                    if dataz[index][var] == 0:
+                        dataz[index][var] = 1
+                dataz[index] = np.log10(dataz[index])
+            index += 1
+    return datax, datay, dataz
+
+
+def plot_2d(
+    source,
+    scan_series,
+    var_series=None,
+    positioner=None,
+    detector=None,
+    monitor=None,
+    log=False,
+    **kwargs,
+):
+    """
+    Plot 2d: Creates 2D plot of scans as function of variable parameter
+
+    Parameters
+    ----------
+    source : databroker database, name of the spec file, or 'csv'
+        Note that applicable kwargs depend on this selection.
+    scan_series : list, int
+        start, stop, step, [start2, stop2, step2, ... ,startn, stopn, stepn]
+        e.g. [10,14,2,23,27,4] will use scan #10,12,14,23,27
+    var_series: string or list
+        string:
+            - Varying variable for scan series to be read from scan (detector),
+                e.g. SampK (sample temperature), optional.
+            - String starting with #metadata, reads metadata from CSV baseline
+        list: Information on metadata to be read: List starting with #P, #U or #Q for
+            motor positions, user values or Q-position, optional:
+            #P: ['#P', row, element_number], e.g. ['#P', 2, 0]
+            #U: ['#U', Variable, element_number], e.g. ['#U', 'KepkoI', 1]
+            #Q: ['#Q', None, element_number], e.g. ['#Q', None, 0]
+        If None, successive scans will be numbered starting from zero.
+    positioner : string, optional
+        Name of the positioner, this needs to be the same as defined in
+        Bluesky or SPEC. If None is passed, it defauts to '4C Theta' motor.
+    detector : string, optional
+        Detector to be read from this scan, again it needs to be the same name
+        as in Bluesky. If None is passed, it defaults to the APD detector.
+    monitor : string, optional
+        Name of the monitor detector for normalization. If None is passed, data are not normalized.
+    log: boolean
+        If True, intensities plotted in logarithmic scale.
+    output: string
+        Output file for png file of plot.
+    kwargs:
+        The necessary kwargs are passed to the loading and fitting functions defined by the
+        `source` argument:
+            - csv        -> possible kwargs: folder, name_format, e.g. "scan_{}_primary.csv"
+            - spec       -> possible kwargs: folder
+            - databroker -> possible kwargs: stream, query
+        Note that a warning will be printed if the an unnecessary kwarg is
+        passed.
+
+    Returns
+    -------
+    2D plot, png-file
+
+    """
+    folder = kwargs.pop("folder", "")
+    output = kwargs.pop("output", None)
+    if isinstance(source, (str, SpecDataFile)) and source != "csv":
```

Comment:
Aren't lines 559 to 576 handled by `load_series`? Maybe these are not needed here?

Reply:
In principle yes, but then plot_2d does not know the positioner for the x-axis label, in case the positioner is taken from default.